src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/transformer/DocumentTransformer.kt

```diff
@@ -26,23 +26,21 @@ class DocumentTransformer {
       )
     }
 
-    val documentsForConvictionId = groupedDocuments.convictions
-      .firstOrNull { it.convictionId == convictionId.toString() }
+    val documentsForConvictionId = groupedDocuments
+      .convictions
+      .filter { convictionId == null || it.convictionId == convictionId.toString() }
```

Comment:
filtering on `convictionId == null` seems like a confusing mechanism!

Would things be easier if we used two different transformer functions, eg. 

- `DocumentTransformer.toApiFiltered(documents, convictionId)` and
- `DocumentTransformer.toApiUnfiltered(documents)`

then the controller could avoid setting `convictionId` to null -- which doesn't help us understand that we're actually interested in **all** convictions!

Reply:
Yes, I completely agree, also the variable being called documentsForConvictionId is quite confusing. I'll make the proposed change before merging